crypto/params.c

```diff
@@ -1529,3 +1529,29 @@ int OSSL_PARAM_get_octet_string_ptr(const OSSL_PARAM *p, const void **val,
     return rv || get_string_ptr_internal(p, val, used_len,
                                          OSSL_PARAM_OCTET_STRING);
 }
+
+OSSL_PARAM *OSSL_PARAM_next(OSSL_PARAM *p)
+{
+    if (p != NULL && p->key != NULL)
+        return p + 1;
+    return NULL;
+}
+
+const OSSL_PARAM *OSSL_PARAM_next_const(const OSSL_PARAM *p)
+{
+    if (p != NULL && p->key != NULL)
+        return p + 1;
+    return NULL;
+}
+
+int OSSL_PARAM_equals(const OSSL_PARAM *p, const char *key)
```

Comment:
OSSL_PARAM is a **public** structure, this can be done directly.
It's also incorrect, param names are case insensitive.

Reply:
OSSL_PARAM_locate performs a case sensitive comparation:

https://github.com/openssl/openssl/blob/878ff0a1ada6411942d04bf98d61747e5527fa9b/crypto/params.c#L53